tooling/nargo/src/ops/test.rs

```diff
@@ -278,85 +266,41 @@ fn check_expected_failure_message(
 }
 
 /// A specialized foreign call executor which tracks whether it has encountered any unknown foreign calls
-struct TestForeignCallExecutor<'a, F> {
-    /// The executor for any [`ForeignCall::Print`] calls.
-    printer: PrintForeignCallExecutor<'a>,
-    mocker: MockForeignCallExecutor<F>,
-    external: Option<RPCForeignCallExecutor>,
-
+struct TestForeignCallExecutor<E> {
+    executor: E,
     encountered_unknown_foreign_call: bool,
 }
 
-impl<'a, F: Default> TestForeignCallExecutor<'a, F> {
-    fn new(
-        output: PrintOutput<'a>,
-        resolver_url: Option<&str>,
-        root_path: Option<PathBuf>,
-        package_name: Option<String>,
-    ) -> Self {
-        let id = rand::thread_rng().gen();
-        let printer = PrintForeignCallExecutor { output };
-        let external_resolver = resolver_url.map(|resolver_url| {
-            RPCForeignCallExecutor::new(resolver_url, id, root_path, package_name)
-        });
-        TestForeignCallExecutor {
-            printer,
-            mocker: MockForeignCallExecutor::default(),
-            external: external_resolver,
-            encountered_unknown_foreign_call: false,
-        }
+impl<E> TestForeignCallExecutor<E> {
+    #[allow(clippy::new_ret_no_self)]
+    fn new<'a, F>(output: PrintOutput<'a>, foreign_call_executor: &F) -> Self
+    where
+        F: Fn(PrintOutput<'a>, layers::Unhandled) -> E + 'a,
+    {
+        // Use a base layer that doesn't handle anything, which we handle in the `execute` below.
+        let executor = foreign_call_executor(output, layers::Unhandled);
+
+        Self { executor, encountered_unknown_foreign_call: false }
```

Comment:
```suggestion
    fn new(executor: &E) -> Self {
        Self { executor, encountered_unknown_foreign_call: false }
```

I don't think there's any benefit of passing in anything other than the internal foreign call executor, right? We can call `foreign_call_executor` within `run test` and just pass it in here.

I'm also skeptical about passing ``layers::Unhandled` as we're doing here, we have no guarantees that `foreign_call_executor` will use the specified `base` layer so we're just relying on convention. What do you think of just documenting this requirement and relying on the user to use this base layer?